### ISSUE TITLE
AttributeError: 'dict' object has no attribute 'iteritems'

### ISSUE THREAD

**Issue by primesoftware** [FontCustom/fontcustom#297]
I tried compiling some SVG files and got the following error. Any ideas?

Traceback (most recent call last):                                                                                                    
File "/home/me/.gem/ruby/2.2.0/gems/fontcustom-1.3.8/lib/fontcustom/scripts/generate.py", line 86, in <module> for glyph, data in manifest['glyphs'].iteritems():
AttributeError: 'dict' object has no attribute 'iteritems' error  `fontforge` compilation failed.
/home/me/.gem/ruby/2.2.0/gems/fontcustom-1.3.8/lib/fontcustom/generator/font.rb:92:in `create_fonts'
/home/me/.gem/ruby/2.2.0/gems/fontcustom-1.3.8/lib/fontcustom/generator/font.rb:20:in `generate'
/home/me/.gem/ruby/2.2.0/gems/fontcustom-1.3.8/lib/fontcustom/base.rb:50:in `start_generators'
/home/me/.gem/ruby/2.2.0/gems/fontcustom-1.3.8/lib/fontcustom/base.rb:22:in `compile'
/home/me/.gem/ruby/2.2.0/gems/fontcustom-1.3.8/lib/fontcustom/cli.rb:77:in `compile'
/home/me/.gem/ruby/2.2.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
/home/me/.gem/ruby/2.2.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
/home/me/.gem/ruby/2.2.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
/home/me/.gem/ruby/2.2.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
/home/me/.gem/ruby/2.2.0/gems/fontcustom-1.3.8/bin/fontcustom:5:in `<top (required)>'
/home/me/.gem/ruby/2.2.0/bin/fontcustom:23:in `load'
/home/me/.gem/ruby/2.2.0/bin/fontcustom:23:in `<main>'

**Comment by grubersjoe** [comment]
I'm also unable to compile on Arch Linux (python3 set as default).

#271 seems related but was fixed. Does someone know what is wrong?

```
File "/home/joe/.gem/ruby/2.3.0/gems/fontcustom-1.3.8/lib/fontcustom/scripts/generate.py", line 86, in <module>
                  for glyph, data in manifest['glyphs'].iteritems():
              AttributeError: 'dict' object has no attribute 'iteritems'
       error  `fontforge` compilation failed.
/home/joe/.gem/ruby/2.3.0/gems/fontcustom-1.3.8/lib/fontcustom/generator/font.rb:92:in `create_fonts'
/home/joe/.gem/ruby/2.3.0/gems/fontcustom-1.3.8/lib/fontcustom/generator/font.rb:20:in `generate'
/home/joe/.gem/ruby/2.3.0/gems/fontcustom-1.3.8/lib/fontcustom/base.rb:50:in `start_generators'
/home/joe/.gem/ruby/2.3.0/gems/fontcustom-1.3.8/lib/fontcustom/base.rb:22:in `compile'
/home/joe/.gem/ruby/2.3.0/gems/fontcustom-1.3.8/lib/fontcustom/cli.rb:77:in `compile'
/home/joe/.gem/ruby/2.3.0/gems/thor-0.19.1/lib/thor/command.rb:27:in `run'
/home/joe/.gem/ruby/2.3.0/gems/thor-0.19.1/lib/thor/invocation.rb:126:in `invoke_command'
/home/joe/.gem/ruby/2.3.0/gems/thor-0.19.1/lib/thor.rb:359:in `dispatch'
/home/joe/.gem/ruby/2.3.0/gems/thor-0.19.1/lib/thor/base.rb:440:in `start'
/home/joe/.gem/ruby/2.3.0/gems/fontcustom-1.3.8/bin/fontcustom:5:in `<top (required)>'
/home/joe/.gem/ruby/2.3.0/bin/fontcustom:23:in `load'
/home/joe/.gem/ruby/2.3.0/bin/fontcustom:23:in `<main>'
```

**Comment by elektronik2k5** [comment]
@primesoftware @grubersjoe did you guys manage to resolve this issue? :sos:

**Comment by grubersjoe** [comment]
I've been examining this issue a bit closer. The error message implies, that the `generate.py` script is called with the `python3` interpreter and I verified this to be true. I tried to enforce `python2` through a shebang, but actually this isn't really the problem. In `lib/fontcustom/generator/font.rb` the following command is executed:

```ruby
cmd = "fontforge -script #{Fontcustom.gem_lib}/scripts/generate.py #{@manifest.manifest}"
```

So at least here on Arch Linux the problem is rather, that `fontforge` uses the `python3` environment by default. Unfortanately it seems like the Arch package maintainer stripped `python2` support entirely:

```bash
~ $ python3 -c "import fontforge"
~ $ python2 -c "import fontforge"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named fontforge
```
After building the fontforge package myself I realized however, that fontforge itself performs a check fort he Python version. So this isn't the maintainers "fault". I changed the default Python interpreter (`/usr/bin/python`) to `python2` and fontforge then actually uses the desired  `python2` module.

```bash
checking for a Python interpreter with version >= 2.7... python
checking for python... /usr/bin/python
checking for python version... 2.7
checking for python platform... linux2
checking for python script directory... ${prefix}/lib/python2.7/site-packages
checking for python extension module directory... ${exec_prefix}/lib/python2.7/site-packages
checking for PYTHON... yes
```
So in conclusion there are two things that can be done:

1. Set your default Python interpreter to `python2` and build fontforge with `makepkg` yourself (cumbersome and ugly)
1. Or what would be a much better solution in my opinion: simply port the generate.py script  to Python 3. However I don't know how @kaizau thinks about this...

**Comment by antgel** [comment]
Bit me as well, are there plans to make generate.py work with Python 3?
